pkg/util/time_cache.go

```diff
@@ -113,13 +119,13 @@ func (c *timeCache) fillOrWait(key string) chan T {
 
 		// Store in cache
 		c.lock.Lock()
+		defer c.lock.Unlock()
 		c.cache[key] = data
-		c.lock.Unlock()
 
 		// Remove in flight entry
 		c.inFlightLock.Lock()
```

Comment:
You deferred `c.inFlightLock.Unlock()` in fillOrWait. Therefore, the anonymous function can't continue until fillOrWait returns.